### PR TITLE
Add deinterlacing shader variants

### PR DESCRIPTION
**Pull request by johnnovak**
# Description

This is considered a fix for the `misc/fixvideo` shader—I went a bit over-eager with the improvements there and forgot about the "integer scaling disabled" use case 😊 

The improved version of the `misc/fixvideo` shader that retains a subtle scanlined look only works correctly with vertical integer scaling enabled. But the old variant worked fine with non-integer scaling ratios, which might be still beneficial in some situations (e.g., for 640x400 games; these games would be upscaled only to 800 vertical lines on 1080p displays, resulting in heavy letterboxing).

This PR renames the improved shader to `misc/fixvideo-scanlines` and restores the old shader under the name`misc/fixvideo`.

# Manual testing

Testing with Hardline (1997). This game uses the 640x400 16-bit high colour VESA mode 111h.

Open all images in new tabs with 100% magnification.

## Upscaled capture

Just to demonstrate how the raw image looks like:

raw

## With `misc/fixvideo` & integer scaling disabled

There are no interference artifacts.

fixvideo

## With `misc/fixvideo-scanlines` & integer scaling disabled

There are interference artifacts. This is OK though as this shader is designed to be used with vertical integer scaling enabled.

fixvideo-scanlines

## With `misc/fixvideo-scanlines` & vertical integer scaling

With vertical integer scaling enabled, the interference patterns are gone.

fixvideo-scanlines-integer

fixvideo-scanlines-integer-video

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: contrib/resources/glshaders/misc/fixvideo.glsl [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+#define SAMPLE_POINT_1 0.00
+#define SAMPLE_POINT_2 0.25
+#define SAMPLE_POINT_3 0.75
+#define SAMPLE_POINT_4 1.00
+
+//   Intensity of the deinterlaced odd & even scanlines.
+//   Use values above 1.0 for extra brightness boost.
+//   1.00 = full intensity, 0.0 = zero intensity (black)
+#define ODD_LINE_INTENSITY  1.00
+#define EVEN_LINE_INTENSITY 1.00
```
Comment (johnnovak) [inline contrib/resources/glshaders/misc/fixvideo.glsl:53]: The only difference between the two shaders is the `ODD_LINE_INTENSITY` and `EVEN_LINE_INTENSITY` settings.

File: contrib/resources/glshaders/misc/fixvideo-scanlines.glsl [hunk 2]
```diff
@@ -0,0 +1,171 @@
+#version 120
+
+/*
+ *  Adaptation of the old fixvideo.fx Daum shader originally written by Ove
+ *  Kavven with some improvements and an intelligent scanline-doubling method
+ *  added by John Novak.
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2023 The DOSBox Team
```
Comment (kcgen) [inline contrib/resources/glshaders/misc/fixvideo-scanlines.glsl:10]: Suggest clarifying and attributing the teams on separate lines (as many as needed). The DOSBox Team will have merges from: https://www.dosbox.com/crew.php

**Review by kcgen — APPROVED** [review]
Looks great, @johnnovak !
(Just one minor comment)

File: contrib/resources/glshaders/misc/fixvideo-scanlines.glsl (hunk 2 above)
Comment (johnnovak) [reply contrib/resources/glshaders/misc/fixvideo-scanlines.glsl]: My mistake, I should've put DOSBox Staging Team there. Fixed.

**Pull request merged**